godot-ffi/src/global.rs

```diff
@@ -137,32 +120,63 @@ impl<T> Global<T> {
             }
         };
 
-        Some(GlobalGuard { mutex_guard })
+        // SAFETY: `mutex_guard` was either set to `Initialized` above, or we returned from the function.
+        Some(unsafe { GlobalGuard::new_unchecked(mutex_guard) })
     }
 }
 
 // ----------------------------------------------------------------------------------------------------------------------------------------------
 // Guards
 
-/// Guard that temporarily gives access to a `Global<T>`'s inner value.
-pub struct GlobalGuard<'a, T> {
-    mutex_guard: MutexGuard<'a, InitState<T>>,
-}
+mod global_guard {
```

Comment:
I don't think this extra module is necessary, the separator is enough 🙂

Reply:
the extra module is there to ensure the `mutex_guard` field isn't accessible outside of a very limited set of functions. of course it isn't technically necessary, but by doing this we keep the safety critical logic more local since anything that wants to construct the `GlobalGuard` must go through either `new` or `new_unchecked`.

Reply:
I see, OK. Could you add this as a comment before `mod`?
Something like `// Encapsulate private fields.` or so is enough.